### DISCUSSION TITLE
Designing the dataset

### DISCUSSION THREAD

**Discussion by aryasuneesh** [lucidrains/musiclm-pytorch#49] [General]
I am currently working on using this implementation for a project. I want to know how the dataset should be designed so that the model can use it as input for training.
- How to pre-process the audio files?
- How to label the data so it can be used for training?
- Can I use HuggingFace's Audio feature to prepare the dataset?

Thanks in advance!

**Comment by aropemaker** [comment]
Hi there, I have not managed to train a model successfully yet, however, I am basing mine off the one used by Google. It's linked in their paper, and here is the link for it:
https://www.kaggle.com/datasets/googleai/musiccaps

and in terms of pre-processing audio, you can find the details here on their paper:
https://arxiv.org/abs/2301.11325

With regards to your third point, yes you can use HuggingFace's Audio feature.

Hope this helps :)

**Reply by aryasuneesh** [reply]
Thank you so much for your help!

**Reply by aropemaker** [reply]
Glad that you've found it of some use!
